src/App.tsx

```diff
@@ -1,222 +1,182 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import {
+  createTodo,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+} from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Footer } from './components/Footer/Footer';
+import { TodoList } from './components/Todolist/Todolist';
+import { Error } from './components/Error/Error';
+import { Todo, TodoTitle } from './types/Todo';
+
+enum FilteredStatus {
+  ALL = 'all',
+  ACTIVE = 'active',
+  COMPLETED = 'completed',
+}
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [filterTodos, setFilterTodos] = useState<string>(FilteredStatus.ALL);
+  const [errorType, setErrorType] = useState('');
+  const [changeTodos, setChangeTodos] = useState(true);
+
+  const showError = (text: string) => {
+    setErrorType(text);
+    setTimeout(() => {
+      setErrorType('');
+    }, 2000);
+  };
+
+  const fetshTodo = async () => {
+    if (query.trim().length === 0) {
+      showError('type something');
+
+      return;
+    }
+
+    try {
+      const newTodo = await createTodo(user?.id, query);
+
+      setTodos((prevState) => {
+        return [...prevState, newTodo];
+      });
+    } catch (error) {
+      showError('Unable to add a todo');
+    }
+  };
+
+  const removeTodo = useCallback(async (todoId: number) => {
+    try {
+      await deleteTodo(todoId);
+      setTodos(prev => prev.filter((x) => x.id !== todoId));
+    } catch (error) {
+      showError('Unable to delete a todo');
+    }
+  }, []);
+
+  const changeTodo = useCallback(async (todoId: number, object: TodoTitle) => {
+    try {
+      const updetedTodo: Todo = await updateTodo(todoId, object);
+
+      setTodos(prev => (prev.map((item) => (item.id === todoId
+        ? updetedTodo
+        : item))
+      ));
+    } catch (error) {
+      showError('Unable to update a todo');
+    }
+  }, []);
+
+  const changeAllTodos = () => {
+    todos.forEach(todo => {
+      changeTodo(todo.id, { completed: changeTodos });
+    });
+
+    setChangeTodos(!changeTodos);
+  };
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
+    getTodos(user?.id)
+      .then(res => setTodos(res))
+      .catch(() => errorType);
```

Comment:
let's not mix `try/catch` and `then/catch` in one app. use `async/await` with `try/catch`